### PR TITLE
created post endpoint for user creation

### PR DESCRIPTION
**Pull request by zhangrmatthew**

### CODE REVIEW

File: back-end/app.js [hunk 1]
```diff
@@ -16,11 +16,51 @@ app.use(cors()); // prevents requests from being blocked by CORS
...
+        const jsonString = JSON.stringify(jsonData, null, 2);
+        fs.writeFile("database.json", jsonString)
+          .then(() => res.json({ username }))
+          .catch((err) => next(err));
+      } catch (err) {
+        next(err);
+      }
+    })
+    .catch((err) => next(err));
+})
```
Comment (kdduan) [inline back-end/app.js:61]: Endpoint logic looks good, but you should send back the userId in the response. That way, the front end can know whether a user is logged in and appropriately render different views.

**Review by kdduan — CHANGES_REQUESTED** [review]
Endpoint logic itself looks good - left a comment on the response it should be sending to the FE. 

Also, for the next iteration you should connect the endpoint here to the user signup page.

File: back-end/app.js (hunk 1 above)
Comment (zhangrmatthew) [reply back-end/app.js:61]: Thanks for the comment - I'm not exactly sure what an userId is, I assumed it was just the email as that was the key in the database for users

Comment (kdduan) [reply back-end/app.js:61]: Yep I think we're just using the user's email as the userID to keep things simple since it's a 1:1 mapping from email to user.

Comment (zhangrmatthew) [reply back-end/app.js:61]: Screenshot from 2021-11-06 11-52-48
This is currently what the JSON looks like, would this be enough for the frontend? Or are you talking about the error message itself

Comment (kdduan) [reply back-end/app.js:61]: Yea that should be enough - the FE just needs some response to know that the user is logged in

Comment (zhangrmatthew) [reply back-end/app.js:61]: Ok - I'll add that

Comment (zhangrmatthew) [reply back-end/app.js:61]: Added - please let me know if good to merge

**Review by kdduan — APPROVED** [review]
Looks good! Thanks for making the changes

**Pull request merged**